src/components/modelsList/ModelsList.tsx

```diff
@@ -0,0 +1,146 @@
+import TextButton from '../TextButton/TextButton';
+import './ModelsList.css';
+
+interface modelItem {
+  id: string;
+  name: string;
+}
+
+const ModelsList = (): JSX.Element => {
+  const dispatch = useAppDispatch();
+  const [modelsList, setModelsList] = useState<modelItem[]>([]);
+  const [selectedModel, setSelectedModel] = useState<modelItem | null>(null);
+  const [colors, setColors] = useState<string[]>([]);
+  const [loadingColors, setLoadingColors] = useState(false);
+  const [openColorDialog, setOpenColorDialog] = useState(false);
+  const [snackbar, setSnackbar] = useState<SnackBarState>(initialSnackBarState);
+
+  useEffect(() => {
+    const fetchModels = async () => {
+      try {
+        const modelsListFetched = await fetchModelsList();
+        setModelsList(modelsListFetched);
+      } catch (error) {
+        console.error('Error fetching models:', error);
+      }
+    };
+    fetchModels();
+  }, []);
+
+  const handleAddModel = async (model: modelItem, color: string) => {
+    try {
+      const url = await fetchGLBUrl(model.id, color);
+      dispatch(addModel(model.id, model.name, color, url));
+      setSnackbar({
+        open: true,
+        message: `Added ${model.name} with color ${color}`,
+        severity: 'success',
+      });
+      setOpenColorDialog(false);
+      setSelectedModel(null);
+    } catch (error) {
+      setSnackbar({
+        open: true,
+        message: 'Error adding model',
+        severity: 'error',
+      });
+    }
+  };
+
+  const handleModelClick = async (model: modelItem) => {
+    setLoadingColors(true);
+    try {
+      const fetchedColors = await fetchModelColors(model.id);
+      setColors(fetchedColors);
+      setSelectedModel(model);
+      setOpenColorDialog(true);
+    } catch (error) {
+      setSnackbar({
+        open: true,
+        message: 'Error fetching model colors',
+        severity: 'error',
+      });
+    } finally {
+      setLoadingColors(false);
+    }
+  };
+
+  return (
+    <div className="models-list-container">
+      <List>
+        {modelsList.map((model) => (
+          <ListItem
+            component="button"
+            key={model.id}
+            onClick={() => handleModelClick(model)}
+            className="list-item"
+          >
+            <ListItemText className="list-item-text" primary={model.name} />
+          </ListItem>
+        ))}
+      </List>
+      <Dialog open={openColorDialog} onClose={() => setOpenColorDialog(false)}>
+        <DialogTitle className="dialog-title">Select a Color</DialogTitle>
+        <DialogContent className="dialog-content">
+          {loadingColors ? (
+            <div className="color-loader">
+              <CircularProgress />
+            </div>
+          ) : (
+            <List>
+              {colors.map((color) => (
+                <ListItem
+                  component="button"
+                  key={color}
+                  onClick={() => handleAddModel(selectedModel!, color)}
```

Comment:
this triggers a warning about a non-null assertion, perhaps refactoring `handleAddModel` to have a nullable parameter and checking for null inside the function would work